services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java

```diff
@@ -423,19 +433,20 @@ private Response getErrorResponse(ErrorType errorType) {
         return Response.status(Response.Status.BAD_REQUEST).entity(errorResponse).build();
     }
 
-    // Return all {@link  OID4VCClient}s that support the given type and format
-    private List<OID4VCClient> getClientsOfType(String vcType, Format format) {
-        LOGGER.debugf("Retrieve all clients of type %s, supporting format %s", vcType, format.toString());
+    // Return all {@link  OID4VCClient}s that support the given scope and format
+    // Scope might be different from vct. In the case of sd-jwt for eaxample
+    private List<OID4VCClient> getClientsOfScope(String vcScope, Format format) {
+        LOGGER.debugf("Retrieve all clients of scope %s, supporting format %s", vcScope, format.toString());
 
-        if (Optional.ofNullable(vcType).filter(type -> !type.isEmpty()).isEmpty()) {
-            throw new BadRequestException("No VerifiableCredential-Type was provided in the request.");
+        if (Optional.ofNullable(vcScope).filter(scope -> !scope.isEmpty()).isEmpty()) {
+            throw new BadRequestException("No VerifiableCredential-Scope was provided in the request.");
```

Comment:
I am afraid that it causes 500 instead of 401. Could you test this point by writing an integration test and run it with `-Pauth-server-quarkus` ?

Reply:
Resolved with [346b41](https://github.com/keycloak/keycloak/commit/346b4151539737e83126f8cc9705b462441c1081)